test/request/hostname.js

```diff
@@ -11,6 +11,12 @@ describe('req.hostname', function(){
     req.hostname.should.equal('foo.com');
   })
 
+  it('should work with IPv6 Host', function(){
+    var req = request();
+    req.header.host = '[::1]:3000';
+    assert.equal(req.hostname, '[::1]');
+  })
+
```

Comment:
What should req.hostname be if req.host is `[::1]` without the port? Or is req.host guaranteed with port attached such as `[::1]:80`? 

Otherwise, if req.host were `[::1]`, this req.hostname would be  `[::`.